### PR TITLE
Added configury option to enable deprecated tests

### PR DESCRIPTION
**Pull request by kholland-intel**

### CODE REVIEW

File: test/shmemx/Makefile.am [hunk 1]
```diff
@@ -78,6 +78,11 @@ gettid_register_LDADD = $(LDADD) $(PTHREAD_CFLAGS)
 
 AM_CPPFLAGS += -DENABLE_SHMEMX_TESTS
 
+if ENABLE_DEPRECATED_TESTS
+AM_CPPFLAGS += -DENABLE_DEPRECATED_TESTS
+endif
+
+
```
Comment (davidozog) [inline test/shmemx/Makefile.am:85]: remove this extra line, por favor.

**Comment by davidozog** [comment]
@kholland-intel - looks great, thanks.

I see there is a test "row" in `<SOS>/.github/workflow/ci.xml` that enables deprecated tests using CFLAGS.  Can you please update that to use the new configure flag?

File: test/shmemx/Makefile.am (hunk 1 above)
Comment (kholland-intel) [reply test/shmemx/Makefile.am]: Fixed 👍

**Comment by kholland-intel** [comment]
> @kholland-intel - looks great, thanks.
> 
> I see there is a test "row" in `<SOS>/.github/workflows/ci.xml` that enables deprecated tests using CFLAGS. Can you please update that to use the new configure flag?

Did you mean `<SOS>/.github/workflows/ci.yml` ? If so, I just added the new configure to the "Deprecated tests" row.

File: test/shmemx/Makefile.am [hunk 2]
```diff
@@ -78,6 +78,10 @@ gettid_register_LDADD = $(LDADD) $(PTHREAD_CFLAGS)
 
 AM_CPPFLAGS += -DENABLE_SHMEMX_TESTS
 
+if ENABLE_DEPRECATED_TESTS
+AM_CPPFLAGS += -DENABLE_DEPRECATED_TESTS
```
Comment (davidozog) [inline test/shmemx/Makefile.am:82]: We should also add this to `AM_CFLAGS` right?

File: test/unit/Makefile.am [hunk 3]
```diff
@@ -208,6 +208,10 @@ AM_CFLAGS = -DENABLE_SHMEMX_TESTS
 AM_CPPFLAGS += -DENABLE_SHMEMX_TESTS
 endif
 
+if ENABLE_DEPRECATED_TESTS
+AM_CFLAGS = -DENABLE_DEPRECATED_TESTS
```
Comment (davidozog) [inline test/unit/Makefile.am:212]: We should also add this to `AM_CPPFLAGS` right?

**Review by davidozog — APPROVED** [review]

**Comment by davidozog** [comment]
Are we good to merge this one @wrrobin and @kholland-intel?

**Review by wrrobin — APPROVED** [review]

**Comment by kholland-intel** [comment]
> Are we good to merge this one @wrrobin and @kholland-intel?

I think we are good to go. We have both approvals so I will go ahead and merge.

**Pull request merged**